File: Java/P4/Civitas/src/civitas/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

/**
 *
 * @author arturo
 */
public class Dado{
    private int random;
    private int ultimoResultado;
    private boolean debug;
    private final Diario event;
    static final private Dado instance=new Dado();
    private static final int SalidaCarcel=5;
    
    static public Dado getInstance() {
        return instance;
    }
    
    private Dado(){ 
        this.random=0;
        this.ultimoResultado=0;
        this.event=Diario.getInstance();
        this.debug=false;
    }
    
    int tirar(){
        if(!debug){
            this.random=(int)(Math.random()*6+1);
            this.ultimoResultado=this.random;
        }else{
            this.random=1;
            this.ultimoResultado=this.random;
        }
       return this.ultimoResultado;
    }
    
    int getUltimoResultado(){
        return this.ultimoResultado;
    }
    
    boolean SalgoDeLaCarcel(){
        boolean salir=false;
        
        if(getUltimoResultado()>=SalidaCarcel){
            salir=true;
        }
    
        return salir;
    }
    
    int quienEmpieza(int n){
        int aleatorio;
        
        aleatorio=(int)(Math.random()*(n-1)+1);
        
        return aleatorio;
    }
    
    public void setDebug(boolean d){ // Public para el main del otro paquete (JuegoTexto)
        this.debug=d;
        String estado;
       if(this.debug==false){
           estado=" Debug desactivado";
           System.out.print("Diario: "+estado+"\n");
           event.ocurreEvento(estado);
       }else{
           estado=" Debug activado";
           System.out.print("Diario: "+estado+"\n");
           event.ocurreEvento(estado);
       } 
    }
}

File: Ruby/Civitas/pkg/Civitas-0.0.1/lib/dado.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

# encoding: utf-8
require 'singleton'
require_relative 'diario'

module Civitas
  class Dado
    include Singleton 
    @random
    @ultimoresultado
    @debug
    @@Salida_carcel=5
    
    def initialize
      @random=Random.new
      @ultimoresultado=0
      @debug=false
    end
    
    attr_reader :Salida_carcel, :debug
    attr_accessor :ultimo_resultado
    
    def tirar
      if(!debug)
        tirada=@random.rand(1..6)
        @ultimoresultado=tirada
      else
        tirada=1
        @ultimoresultado=1
      end
      
      return tirada
    end
    
    def salgo_de_la_carcel
      aux=false
      
      if(ultimo_resultado>=Salida_carcel)
        aux=true
      end
      
      return aux
    end
    
    def quien_empieza(n)
      jugador_aleatorio=@random.rand(n)
      return jugador_aleatorio
    end
    
    def set_debug(db)
      @debug=db
      estado=" "
      
      if(!debug)
        estado="Debug desactivado"
        Diario.instance.ocurre_evento(estado)
      else
        estado="Debug activado"
        Diario.instance.ocurre_evento(estado)
      end
    end
  end
end
